### PR TITLE
Ensure null-termination of string in negotiate_wrapper

### PR DESCRIPTION
**Pull request by kinkie**
Coverity identified a theoretical chance that a buffer may not be
null-terminated in negotiate_wrapper. The code flow is clean, adding a
forced null termination to apply defensive programming practices.

### CODE REVIEW

File: src/auth/negotiate/wrapper/negotiate_wrapper.cc [hunk 1]
```diff
@@ -257,6 +257,7 @@ processingLoop(FILE *FDKIN, FILE *FDKOUT, FILE *FDNIN, FILE *FDNOUT)
                 return 0;
             }
         }
+        buff[sizeof(buff)-1] = '\0';
```
Comment (rousskov) [inline src/auth/negotiate/wrapper/negotiate_wrapper.cc:260]: Like you said, the official code probably does terminate `buff` correctly so this proposed code causes no harm, but if we ignore the well-hidden "already terminated correctly" fact, then this proposed code looks as if it silently truncates input! We can avoid that (false) implication by doing something like this:

```suggestion
        buff[sizeof(buff)-1] = '\0';
        assert(strlen(buff) < sizeof(buf)-1); // the previous assignment did not truncate
```

or this


```suggestion
        buff[sizeof(buff)-1] = '\0'; // paranoid; already terminated correctly
```

I do not insist on this change.

**Review by rousskov — COMMENTED** [review]
If at all possible, please cite the exact Coverity defect ID(s) (with the corresponding defect gists) in the PR description like we did in commits 5d84beb, 9fa45f8, 699ade2. I would not claim "resolves" like we did in commits like ca0d7ed unless you have actually _tested_ that this change resolved that Coverity defect.

File: src/auth/negotiate/wrapper/negotiate_wrapper.cc (hunk 1 above)
Comment (rousskov) [reply src/auth/negotiate/wrapper/negotiate_wrapper.cc]: Thank you for addressing this concern.

**Comment by yadij** [comment]
Passed 10 days with no objections.

**Pull request closed without merging**